zap/src/irgen/des.rs

```diff
@@ -14,8 +14,20 @@ impl Gen for Des {
 		self.buf.push(stmt);
 	}
 
-	fn gen(mut self, var: Var, ty: &Ty) -> Vec<Stmt> {
-		self.push_ty(ty, var);
+	fn gen(mut self, var: Var, types: &[Ty]) -> Vec<Stmt> {
+		for (i, ty) in types.iter().enumerate() {
+			let var = if i > 0 {
+				Var::Name(match &var {
+					Var::Name(name) => format!("{name}{}", i + 1),
+					_ => unreachable!(),
+				})
+			} else {
+				var.clone()
+			};
+
+			self.push_ty(ty, var);
```

Comment:
This is... awkward.
Let's instead add an `nsuffix` method to the `Var` impl (see `var.nindex` as a good example).
And then we can do something like
```suggestion
			self.push_ty(ty, var.nsuffix(i + 1));
```